tmail-backend/tmail-third-party/openpaas/src/test/java/com/linagora/tmail/carddav/CardDavCreationSerializerTest.java

```diff
@@ -0,0 +1,89 @@
+package com.linagora.tmail.carddav;
+
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.james.core.MailAddress;
+import org.junit.jupiter.api.Test;
+
+import net.javacrumbs.jsonunit.core.Option;
+
+public class CardDavCreationSerializerTest {
+
+    CardDavCreationSerializer testee = new CardDavCreationSerializer();
+
+    @Test
+    void testSerializeWithAllFields() throws Exception {
+        // Arrange
+        CardDavCreationObjectRequest.Email email = new CardDavCreationObjectRequest.Email(
+            List.of(CardDavCreationObjectRequest.EmailType.HOME),
+            new MailAddress("anbach4@lina.com")
+        );
+
+        CardDavCreationObjectRequest request = new CardDavCreationObjectRequest(
+            "4.0",
+            "123",
+            Optional.of("An Bach4"),
+            Optional.of(List.of("An", "Bach4")),
+            email
+        );
+
+        // Act
+        String jsonResult = testee.serializeAsString(request);
+
+        // Assert
+        String expectedJson = """
+            [
+               "vcard",
+               [
+                 [ "version", {}, "text", "4.0" ],
+                 [ "uid", {}, "text", "123" ],
+                 [ "fn", {}, "text", "An Bach4" ],
+                 [ "n", {}, "text", [ "An", "Bach4" ] ],
+                 [ "email", { type: [ "Home" ] }, "text", "mailto:anbach4@lina.com" ]
+               ],
+               []
+             ]
+            """;
```

Comment:
No no no no no

Do not use the modified JSON non standard calls that you got from inspecting what the webmail does.

I want real CardDav, compatible with ANY CardDav server, not just our today modified Sabre server.

That way integrating 3rd party address books will be trivial.

Reply:
In ticket
```
Step 3: see if the contact exists: GET https://dav.linagora.com/addressbooks/{_id}/collected/{hex(sha1(mailAddress))}.vcf
```
It's a modified endpoint, not standard. I'm unsure when a "standard" body request, it will work or not. 
Let me study

Reply:
updated with standard body request